OpenEdXMobile/src/main/java/org/edx/mobile/view/MyCoursesListFragment.kt

```diff
@@ -283,11 +251,27 @@ class MyCoursesListFragment : OfflineSupportBaseFragment(), RefreshListener {
         refreshOnResume = true
     }
 
+    @Subscribe
+    fun onEventMainThread(event: IAPFlowEvent) {
+        if (!this.isResumed) {
+            return
+        }
+        when (event.flowAction) {
+            IAPFlowData.IAPAction.SHOW_FULL_SCREEN_LOADER -> {
+                event.iapFlowData?.let {
+                    showFullscreenLoader(event.iapFlowData)
+                }
+            }
+            IAPFlowData.IAPAction.PURCHASE_FLOW_COMPLETE -> {
+                onRefresh()
```

Comment:
IMO, we should refresh the courses directly instead of Posting another Event.

Reply:
no change is required.